satellite/audit/service.go

```diff
@@ -72,6 +78,31 @@ func (service *Service) Run(ctx context.Context) (err error) {
 	})
 }
 
+// RemoteSegment takes a remote segment found in metainfo and creates a reservoir for it if it doesn't exist already
+func (service *Service) RemoteSegment(ctx context.Context, path storj.Path, pointer *pb.Pointer) (err error) {
+	defer mon.Task()(&ctx, path)(&err)
+
+	remote := pointer.GetRemote()
+	pieces := remote.GetRemotePieces()
+
+	for i, piece := range pieces {
+		if _, ok := service.Reservoirs[piece.NodeId]; !ok {
+			reputable, err := service.overlay.IsVetted(ctx, piece.NodeId)
+			if err != nil {
+				service.log.Error("error finding if node is vetted", zap.Error(err))
+				return nil
+			}
+			reservoir := NewReservoir(reputable, service.reservoirConfig)
+			reservoir.Segments[0] = remote
+			service.Reservoirs[piece.NodeId] = reservoir
+		} else {
+			service.Reservoirs[piece.NodeId].sample(remote, i)
```

Comment:
don't do this inside an `else`. We want to sample the piece whether we are creating a reservoir for the first time or using an already created one.

Reply:
This `i` is also going to be the index of the piece, not the total number of pieces seen by the reservoir so far. `sample` should take in the remote segment, but the reservoir itself should determine the count of the items seen so far. I will add a comment in the relevant section of the reservoir code.

Reply:
oh oops, thanks

Reply:
wait actually- re: your first comment, I think it still needs to happen in the `else`, because the slots need to fill up to the reservoir size in the order that they appear in the stream and THEN be randomly switched out.

Reply:
Well `sample()` already deals with that logic, doesn't it? First it checks to see if more slots need to be filled, and if not, it does the random switching.

Reply:
okay I see- yeah will switch things around a bit